game-core/src/main/java/games/strategy/triplea/attachments/AbstractRulesAttachment.java

```diff
@@ -108,7 +104,7 @@ private void setObjectiveValue(final String value) {
   }
 
   @GameProperty(xmlProperty = true, gameProperty = true, adds = false)
-  private void setObjectiveValue(final Integer value) {
+  private void setObjectiveValue(final int value) {
```

Comment:
Just noting that switching to primitives will trigger more null analysis warnings if we ever enable a formal static null analysis engine.  We eventually need to think about how we're going to handle this because it looks like the contract of `MutableProperty#setValue()` is going to have to allow incoming `null` values due to how most `DynamicallyModifiable` classes are written.

Reply:
@ssoloff I think the easiest solution is probably going to be to extend the MutableProperty class with Boolean and Integer types and add a precondition check there